README.md

```diff
@@ -160,7 +160,9 @@ Dalli::Client accepts the following options. All times are in seconds.
 
 **failover**: Boolean, if true Dalli will failover to another server if the main server for a key is down.  Default is true.
 
-**threadsafe**: Boolean.  If true Dalli ensures that only one thread is using a socket at a given time.  Default is true.  Set to false at your own peril.
+**threadsafe**: Boolean.  If true Dalli ensures that only one thread is using a socket at a given time.  Default is true.  Set to false at your own peril. It will be ignored and set to false if `pool_size` option is specified.
```

Comment:
It makes no difference as the store clearly overwrites the option with false -https://github.com/petergoldstein/dalli/blob/master/lib/active_support/cache/dalli_store.rb#L72

Reply:
@j4rs yes, I noticed this as well. (btw, pro-tip, before pasting the github link, hit `y` on your keyboard and the url will change to a ref that is future proof https://github.com/petergoldstein/dalli/blob/8b19276d27d29b7b3abb6e3aab48ce054d165619/lib/active_support/cache/dalli_store.rb#L72 )

let me know if I should make an issue. Here's my experience:

Using Rails with cache_store set to dalli_store, so using the ActiveSupport::Cache::DalliStore
(Also backing rack::attack and the session store with the same cache store and cache server)

we set a pool size per https://www.memcachier.com/documentation/ruby-puma-webserver

```ruby
    concurrency = Integer(ENV['WEB_CONCURRENCY'] || 2)
    threads_count = Integer(ENV['RAILS_MAX_THREADS'] || 5)
    per_dyno_concurrency = concurrency * threads_count
```

As you know, this config generates a dao of a a ConnectionPool (which is supposed to be threadsafe) wrapping an explicitly non-threadsafe Dalli::Client

https://github.com/petergoldstein/dalli/blob/8b19276d27d29b7b3abb6e3aab48ce054d165619/lib/active_support/cache/dalli_store.rb#L69-L73

I added some code to one of our Rails controllers with some behavior using `with` and `multi`

```ruby
Rails.cache.with do |client| 
  client.multi do 
    # various actions 
  end 
end
```

we began experiencing two classes of errors

1. the cache store was retrieving the wrong values from the cache server (whether because they were set incorrectly or retrieved incorrectly, I don't know)
1. we got *a lot* of `DalliError: No server available ` (specifically a `Dalli::RingError`)

After changing the cache server to a new server for both rack::attack and Rails.cache and removing the `with` and `multi` code and deploying to new servers, the problems went away.  I added all the code back and except that I replaced the `with` and `multi` calls with regular `Rails.cache.read/write/fetch`.  So, it seems that the combination of a `pool_size` setting with calling `with/multi` is *not* threadsafe.

Interestingly, the [Rails mem_cache_store](https://github.com/rails/rails/blob/5-0-stable/activesupport/lib/active_support/cache/mem_cache_store.rb) doesn't allow direct access to the dalli::client and doesn't use `multi`.